server/api/crud/workflows.py

```diff
@@ -341,6 +343,18 @@ def _prepare_run_object_for_single_run(
                 mlrun.model.Notification.from_dict(notification.dict())
                 for notification in workflow_request.notifications or []
             ]
+            default_notification_params = (
+                server.api.utils.notification_pusher.get_notifications_default_params()
+            )
+            for notification in notifications:
+                notification_type = NotificationTypes(
+                    notification.kind or NotificationTypes.console
+                ).get_notification()
+
+                default_params = default_notification_params.get(notification.kind, {})
+                notification.params = notification_type.enrich_default_params(
+                    notification.params, default_params
```

Comment:
1.  It is server side code
2. [Here](https://github.com/roei3000b/mlrun/blob/08344c885833faa0464afc4da34f1a9656716d96/server/api/crud/workflows.py#L342-L345) we create notification with `from_dict`, meaning that the `__init__` of the object isn't called, and because of that I need to manually call to the enrichment here.